src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/AIServiceManager.cs

```diff
@@ -51,98 +53,110 @@ private String GetCvServiceBaseUri()
 
         public async Task PostCriteriaImageMatch(CVImageCriteriaRequest request, Action<Action> abortRegistrationHook, Action<List<CVImageResult>> onSuccess, Action onFailure)
         {
-            await SendWebRequest(
-                uri: $"{GetCvServiceBaseUri()}/criteria-image-match",
-                method: "POST",
-                payload: request.ToJsonString(),
-                abortRegistrationHook: abortRegistrationHook.Invoke,
-                onSuccess: (s) =>
-                {
-                    var response = JsonConvert.DeserializeObject<CVImageResultList>(s);
-                    onSuccess.Invoke(response.results);
-                },
-                onFailure: (f) =>
-                {
-                    RGDebug.LogWarning($"Failed to evaluate CV Image criteria: {f}");
-                    onFailure.Invoke();
-                }
-            );
+            if (RGServiceManager.GetInstance().LoadAuth(out var authToken))
```

Comment:
If they are not authed, what happens? It seems like neither action gets called which could cause hanging somewhere? Maybe we should either throw an exception or we could call onFailure. But at a minimum I'd like the user to at least see a message somewhere that they can't use the AI API without being logged in.

Reply:
LoadAuth internally logs a warning if the APIKey isn't configured, same as it does for the existing RGService calls.  If an invalid key is configured, you will get 401 errors from the AIserver calls.